cognite/client/utils/_json.py

```diff
@@ -41,3 +43,31 @@ def dumps(
 
 
 loads = json.loads
+
+# As opposed to protobuf, datapoints in JSON returns out-of-range float values as strings. This means
+# we're forced to do a lookup for every single datapoint we try to insert (and read in retrieve_latest
+# when not ignoring bad)... so we allow some ugly optimizations in the translation code:
+_FLOAT_API_MAPPING = MappingProxyType({"Infinity": math.inf, "-Infinity": -math.inf, "NaN": math.nan})
+_FLOAT_API_MAPPING_REVERSE = MappingProxyType({math.inf: "Infinity", -math.inf: "-Infinity", math.nan: "NaN"})
```

Comment:
Why do you need this? Is it just for the `.dump()` function? I thought you used protobuf both for ingestion and retrieval.

Reply:
`retrieve_latest` and insertion is done using JSON

Reply:
To avoid having to rewrite even more logic... 😬